mesonbuild/interpreterbase/baseobjects.py

```diff
@@ -102,13 +122,20 @@ def method_call(
                 args: T.List[TYPE_var],
                 kwargs: TYPE_kwargs
             ) -> TYPE_var:
-        if method_name in self.methods:
-            method = self.methods[method_name]
+        if method_name in self.METHODS:
+            method = self.METHODS[method_name]
             if not getattr(method, 'no-args-flattening', False):
                 args = flatten(args)
             if not getattr(method, 'no-second-level-holder-flattening', False):
                 args, kwargs = resolve_second_level_holders(args, kwargs)
-            return method(args, kwargs)
+            return method(self, args, kwargs)
+        if method_name in self.methods:
+            bound_method = self.methods[method_name]
+            if not getattr(bound_method, 'no-args-flattening', False):
+                args = flatten(args)
+            if not getattr(bound_method, 'no-second-level-holder-flattening', False):
+                args, kwargs = resolve_second_level_holders(args, kwargs)
+            return bound_method(args, kwargs)
```

Comment:
This logic duplication feels slightly unfortunate. Do you think it makes sense to maybe look up the method from both dictionaries in turn, and then enter the main body just once?

Reply:
I can just change all objects to use the new mechanism and remove the old one (possibly in a separate PR).

Reply:
Did that and removed the old way, since the modules are handled differently and need not be converted at the same time.